### DISCUSSION TITLE
Create an option to serve no client-side javascript to the end user.

### DISCUSSION THREAD

**Discussion by kenmueller** [vercel/next.js#49544] [Ideas]
### Goals

1. People (including me) wanting to make a purely static homepage, or complete website.
2. Support this feature while using the app directory
3. Support server actions
4. Everything must be a server component

### Background

SvelteKit has an option where you set `router = false` and `hydrate = false` to serve no client-side js for that page. Why doesn't Next.js have that feature? `unstable_runtimeJS` was a thing, but that doesn't work for the app directory.

### Proposal

I am interested in contributing. I think that this should be an export on a per-page basis such as `export const hydrate = false`.

**Comment by adanperez** [comment]
I too am looking for a solution similar to `unstable_runtimeJS` for Pages router.
